Description: Upstream changes introduced in version 0.33~cvs20110710-3
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 dietlibc (0.33~cvs20110710-3) experimental; urgency=low
 .
   * Update to cvs20110918.
     - Fixes memchr('\0') (Closes: #582654)
   * SIGILL in sigsetjmp on s390: fixed on last release (Closes: #523110)
 .
 The person named in the Author field signed this changelog entry.
Author: Hector Oron <zumbi@debian.org>
Bug-Debian: http://bugs.debian.org/523110
Bug-Debian: http://bugs.debian.org/582654

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- dietlibc-0.33~cvs20110710.orig/i386/memchr.S
+++ dietlibc-0.33~cvs20110710/i386/memchr.S
@@ -51,7 +51,7 @@ memchr:
 
 .Lfound:
 	cmp %edi,%ecx
-	jb .Lnull
+	jbe .Lnull
 	mov %edi,%eax
 	pop %ebx
 	pop %esi
--- /dev/null
+++ dietlibc-0.33~cvs20110710/parisc/strstr.S
@@ -0,0 +1,34 @@
+.level 1.1
+
+.text
+.globl strstr
+strstr:
+	.PROC
+	.CALLINFO
+	.ENTRY
+	ldb,ma 1(%r25), %r20
+	ldb,ma 1(%r26), %r19
+	ldi 2, %ret0
+.L1:
+	cmpb,=,n %r0, %r19, .Lret2
+	cmpb,<> %r20, %r19, .L1
+	ldb,ma 1(%r26), %r19
+	copy %r19, %r24
+	copy %r25, %r23
+	ldb,ma 1(%r23), %r21
+	copy %r26, %r22
+.Lloop:
+	cmpb,<> %r24, %r21, .L1
+	ldb,ma 1(%r22), %r24
+	cmpb,= %r0, %r24, .Lret
+	ldb,ma 1(%r23), %r21
+	cmpb,<>,n %r21, %r0, .Lloop
+	nop
+.Lret:
+	addi 0, %r26, %ret0
+.Lret2:
+	bv %r0(%rp)
+	addi -2, %ret0, %ret0
+	.EXIT
+	.PROCEND
+	.EXPORT strstr
--- dietlibc-0.33~cvs20110710.orig/test/memchr.c
+++ dietlibc-0.33~cvs20110710/test/memchr.c
@@ -5,6 +5,7 @@ int main() {
   const char* test="blubber";
   assert(memchr("aaaa",'x',4)==0);
   assert(memchr(0,'x',0)==0);
+  assert(memchr(test,'u',2) == NULL);
   assert(memchr(test,'u',7) == test+2);
   assert(memchr(test,'b',7)==test);
   assert(memchr(test+6,'r',1)==test+6);
